infrastructure-playbooks/lv-teardown.yml

```diff
@@ -0,0 +1,103 @@
+- name: tear down existing osd filesystems then logical volumes, volume groups, and physical volumes
+  hosts:
+  - osds
+
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to tear down the logical volumes?
+      default: 'no'
+      private: no
+
+  tasks:
+  - name: exit playbook, if user did not mean to tear down logical volumes
+    fail:
+      msg: >
+        "Exiting lv-teardown playbook, logical volumes were NOT torn down.
+         To tear down the logical volumes, either say 'yes' on the prompt or
+         or use `-e ireallymeanit=yes` on the command line when
+         invoking the playbook"
+    when: ireallymeanit != 'yes'
+
+  - name: include vars of lv_vars.yaml
+    include_vars:
+      file: lv_vars.yaml
+
+  # need to check if lvm2 is installed
+  - name: install lvm2
+    package:
+      name: lvm2
+      state: present
+
+# BEGIN TEARDOWN
+  - name: find any existing OSD filesystems
+    shell: "grep /var/lib/ceph/osd /proc/mounts | awk '{print $2}'"
+    register: old_osd_filesystems
+
+  - name: tear down any existing OSD filesystems
+    shell: "umount -v {{item}}"
+    with_items: "{{ old_osd_filesystems.stdout_lines }}"
+
+  - name: kill all LVM commands that may have been hung
+    shell: "killall -q lvcreate pvcreate vgcreate lvconvert || echo -n"
+    failed_when: false
+
+  ## Logcal Vols
+  - name: tear down existing LV for bucket index
+    lvol:
+      lv: "{{ item.lv_name }}"
+      vg: "{{ nvme_vg_name }}"
+      state: absent
+      force: yes
+    with_items:
+      - "{{ nvme_device_lvs }}"
+
+  - name: tear down any existing hdd data LVs 
+    lvol:
+      lv: "{{ hdd_lv_prefix }}-{{ item.split('/')[-1] }}"
+      vg: "{{ hdd_vg_prefix }}-{{ item.split('/')[-1] }}"
+      state: absent
+      force: yes
+    with_items:
+      - "{{ hdd_devices }}"
+
+  - name: tear down any existing LV of journal for bucket index
+    lvol:
+      lv: "{{ item.journal_name }}"
+      vg: "{{ nvme_vg_name }}"
+      state: absent
+      force: yes
+    with_items:
+      - "{{ nvme_device_lvs }}"
+
+  - name: tear down any existing LVs of HDD journals
+    lvol:
+      lv: "{{ hdd_journal_prefix }}-{{ item.split('/')[-1] }}"
+      vg: "{{ nvme_vg_name }}"
+      state: absent
+      force: yes
+    with_items:
+      - "{{ hdd_devices }}"
+
+  ## Volume Groups
+  - name: Remove VG on nvme device
+    lvg:
+      vg: "{{ nvme_vg_name }}"
+      state: absent
+      force: yes
+
+  - name: Remove VG for each hdd device
+    lvg:
+      vg: "{{ hdd_vg_prefix }}-{{ item.split('/')[-1] }}"
+      state: absent
+      force: yes
+    with_items:
+      - "{{ hdd_devices }}"
+
+  ## Physical Vols
+  - name: tear down PV for nvme device
+    shell: "pvremove --force --yes {{nvme_device}}"
```

Comment:
`{{nvme_device}}` -> `{{ nvme_device }}`

Reply:
@leseb did you mean we replace `shell:` -> `command:` everywhere?

Reply:
Yes, everywhere when no pipe is involved.